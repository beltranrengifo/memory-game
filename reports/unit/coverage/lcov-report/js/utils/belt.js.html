
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for js/utils/belt.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">js/utils</a> belt.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">const belt = '&lt;svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 148 200"&gt;&lt;g&gt;&lt;g&gt;&lt;path d="M148,100A74,74,0,0,0,48,30.73V0H0V200H48V169.27A73.55,73.55,0,0,0,74,174a26,26,0,0,1,26,26h48a73.72,73.72,0,0,0-19.5-50A73.72,73.72,0,0,0,148,100ZM48,100a26,26,0,1,1,26,26A26,26,0,0,1,48,100Z"/&gt;&lt;path d="M0,100v74.88A105.46,105.46,0,0,0,40.9,200H48V169.27h0A74.1,74.1,0,0,1,0,100Z"/&gt;&lt;/g&gt;&lt;/g&gt;&lt;/svg&gt;'
&nbsp;
export {
  belt
}
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Fri Apr 10 2020 11:59:33 GMT+0200 (GMT+02:00)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    