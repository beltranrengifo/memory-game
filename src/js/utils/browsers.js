const setBrowserInDom = () => {
  const classes = []
  if (navigator.userAgent.toLowerCase().indexOf('firefox') > -1) {
    classes.push('firefox')
  }
  document.querySelector('body').classList.add(...classes)
}

export {
  setBrowserInDom
}
