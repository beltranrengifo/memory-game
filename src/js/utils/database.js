import * as firebase from 'firebase/app'
import 'firebase/database'

const firebaseConfig = {
  apiKey: process.env.FIREBASE_API_KEY,
  authDomain: process.env.FIREBASE_AUTH_DOMAIN,
  databaseURL: process.env.FIREBASE_DATABASE_URL,
  projectId: 'onesait-memory-game',
  storageBucket: 'onesait-memory-game.appspot.com',
  messagingSenderId: '854342630585',
  appId: process.env.FIREBASE_APP_ID
}

console.log(process.env.FIREBASE_API_KEY)

firebase.initializeApp(firebaseConfig)
const database = firebase.database()

export {
  database
}
