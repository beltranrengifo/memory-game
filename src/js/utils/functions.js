const capitalize = (value) => {
  if (!value) {
    return ''
  }
  value = value.toString()
  return value.charAt(0).toUpperCase() + value.slice(1)
}

const deKebabCase = (value) => {
  if (!value) {
    return ''
  }
  return value.replace(/-/g, ' ')
}

export {
  capitalize,
  deKebabCase
}
