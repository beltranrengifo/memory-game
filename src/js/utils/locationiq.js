const token = '31a0aa2ee4220c'
const geolocationUrl = (lat, long) => (
  `https://eu1.locationiq.com/v1/reverse.php?key=${token}&lat=${lat}&lon=${long}&format=json`
)

export {
  geolocationUrl
}
