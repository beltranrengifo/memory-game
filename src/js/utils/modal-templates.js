import { capitalize } from '@/js/utils/functions'

const finishGame = results => (`
  <div class="modal__body-wrap">
    <div class="modal__results">
      <h2 class="modal__title">
        Great game!
        <span id="play-again" class="button">Play again</span>
      </h2>
      <div class="modal__row">
        Attemps: <span class="modal__result">${results.attempts}</span>
      </div>
      <div class="modal__row">
        Time: <span class="modal__result">${results.time}</span>
      </div>
      <div class="modal__row">
        Level: <span class="modal__result">${capitalize(results.level)}</span>
      </div>
      <div class="modal__row modal__row--total">
        Total: <span class="modal__result modal__result--total">${results.total}</span>
      </div>
    </div>
    <div class="modal__ranking">
      <div class="modal__row modal__row--ranking">
        <h3 class="modal__ranking-title">Ranking</h3>
        <table class="table table--ranking">
          ${results.ranking}
        </table>
      </div>
    </div>
  </div>
`)

const getRankingTable = (list, levels) => {
  console.log(list)
  const header = `
    <tr>
      <th>User</th>
      <th>Location</th>
      <th>Level</th>
      <th>Attempts</th>
      <th>Time</th>
      <th>Total</th>
    </tr>
  `

  const rows = list.reduce((str, item, i, a) => {
    let el = `<tr class="table__row ${i < 5 ? 'top-five' : i >= a.length - 5 ? 'last-five' : ''}">`
    el += `<td class="table__item table__item--user"><span class="table__item-position">${i + 1}. </span>${capitalize(item.user)}</td>`
    el += `<td class="table__item table__item--location">${item.userLocation && item.userLocation.geocode.address.city}</td>`
    el += `<td class="table__item table__item--level">${capitalize(levels[item.level])}</td>`
    el += `<td class="table__item table__item--attempts">${item.attempts}</td>`
    el += `<td class="table__item table__item--time">${item.time}</td>`
    el += `<td class="table__item table__item--total">${item.total}</td>`
    el += '</tr>'
    str += el
    return str
  }, '')

  /* const rows = list.reduce((str, item, i, a) => {
    let el = `<li class="modal__ranking-list-item ${i < 5 ? 'top-five' : i >= a.length - 5 ? 'last-five' : ''}">`
    el += `<span class="list-item list-item--user">${item.user}</span>`
    el += `<span class="list-item list-item--location">${item.userLocation && item.userLocation.geocode.address.city}</span>`
    el += `<span class="list-item list-item--level">${capitalize(levels[item.level])}</span>`
    el += `<span class="list-item list-item--attempts">${item.attempts}</span>`
    el += `<span class="list-item list-item--time">${item.time}</span>`
    el += `<span class="list-item list-item--total">${item.total}</span>`
    el += '</li>'
    str += el
    return str
  }, '') */

  return `
    <thead class="table__header">
      ${header}
    </thead>
    <tbody class="table__body">
      ${rows}
    </tbody>
  `
}

const getUser = () => (`
  <div
    id="user-modal"
    class="modal__body-wrap">
    <h2 class="modal__title">
      Hi there!
    </h2>
    <div class="modal__row">
      <input
        class="input"
        name="user"
        id="input-user"
        placeholder="Your name">
    </div>
    <div class="modal__footer">
      <button
        class="button">
        Next
      </button>
    </div>
  </div>
`)

const getLocation = () => (`
  <div
    id="location-modal"
    class="modal__body-wrap">
    <h2 class="modal__title">
      Geolocation API!
    </h2>
    <div class="modal__row">
      We collect your location, just for educational purposes
    </div>
    <div class="modal__footer">
      <button
        value="true"
        class="button">
        Ok
      </button>
      <button
        value="true"
        class="button">
        No
      </button>
    </div>
  </div>
`)

const getGameLevel = (user, city) => (`
  <div
    id="level-modal"
    class="modal__body-wrap">
    <h2 class="modal__title">
      Hi, ${capitalize(user)} ${city ? 'from ' + city + ',' : ''} please choose your level!
    </h2>
    <div class="modal__row">
      Higher levels give higher ranks!
    </div>
    <div class="modal__row">
      <div class="modal__levels">
        <button class="levels__level levels__level--easier" value="2">Easier</button>
        <button class="levels__level levels__level--easy" value="3">Easy</button>
        <button class="levels__level levels__level--medium" value="4">Medium</button>
        <button class="levels__level levels__level--hard" value="5">Hard</button>
        <button class="levels__level levels__level--hardest" value="6">Hardest</button>
      </div>
    </div>
  </div>
`)

export {
  finishGame,
  getUser,
  getLocation,
  getGameLevel,
  getRankingTable
}
