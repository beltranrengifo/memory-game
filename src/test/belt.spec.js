import { belt } from '../js/utils/belt'

describe('Belt logo', () => {
  test('belt to be defined', () => {
    expect(belt).toBeDefined()
  })

  test('belt is a string', () => {
    expect(typeof belt).toBe('string')
  })

  test('belt should match snapshot', () => {
    expect(belt).toMatchSnapshot()
  })
})
