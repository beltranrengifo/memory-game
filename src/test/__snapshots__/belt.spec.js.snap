// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Belt logo belt should match snapshot 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 148 200\\"><g><g><path d=\\"M148,100A74,74,0,0,0,48,30.73V0H0V200H48V169.27A73.55,73.55,0,0,0,74,174a26,26,0,0,1,26,26h48a73.72,73.72,0,0,0-19.5-50A73.72,73.72,0,0,0,148,100ZM48,100a26,26,0,1,1,26,26A26,26,0,0,1,48,100Z\\"/><path d=\\"M0,100v74.88A105.46,105.46,0,0,0,40.9,200H48V169.27h0A74.1,74.1,0,0,1,0,100Z\\"/></g></g></svg>"`;
